kernel/src/arch/syscalls.c

```
@@ -943,4 +953,94 @@ int32_t syscall_fork(struct regs *regs)
+    for (uint32_t i = 0; i < pages; i++) {
+        void *virt = (void *)(addr + i * BLOCK_SIZE);
+
+        void *phys = alloc_blocks(1);
+        if (!phys) return -1;
+        memset((void *)((uint32_t)phys + KERNEL_VIRTUAL_BASE), 0, BLOCK_SIZE);
+        map_address_user(virt, phys);
+    }
+
+    /* Register VMA */
+    vma_t *new_vma = malloc(sizeof(vma_t));
+    new_vma->start = addr;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)